internal/storage/v2/clickhouse/tracestore/reader.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package tracestore
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ClickHouse/clickhouse-go/v2/lib/driver"
+
+	"github.com/jaegertracing/jaeger/internal/storage/v2/clickhouse/tracestore/dbmodel"
+)
+
+const (
+	// TODO: revisit query after materialized views are implemented
+	// to see if the DISTINCT is still required
```

Comment:
I think DISTINCT shouldn't hurt either way. If it's already maintained like that by the materialized view then it will be no-op.